stan/math/rev/fun/singular_values.hpp

```diff
@@ -0,0 +1,50 @@
+#ifndef STAN_MATH_REV_FUN_SINGULAR_VALUES_HPP
+#define STAN_MATH_REV_FUN_SINGULAR_VALUES_HPP
+
+#include <stan/math/rev/meta.hpp>
+#include <stan/math/rev/core.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/prim/err/check_nonzero_size.hpp>
+#include <Eigen/SVD>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return the singular values of the specified matrix.
+ *
+ * Adjoint update equation comes from Equation (4) in Differentiable Programming
+ * Tensor Networks(H. Liao, J. Liu, et al., arXiv:1903.09650).
+ *
+ * @tparam EigMat type of input matrix
+ * @param m MxN input matrix
+ * @return Singular values of matrix
+ */
+template <typename EigMat, require_eigen_matrix_dynamic_t<EigMat>* = nullptr,
+          require_vt_var<EigMat>* = nullptr>
+inline auto singular_values(const EigMat& m) {
+  using ret_type = promote_scalar_t<var, Eigen::VectorXd>;
```

Comment:
```
using ret_type = return_var_matrix_t<Eigen::VectorXd, EigMat>;
```

This computes the autodiff return type for a non-autodiff type `Eigen::VectorXd` given the input argument is `EigMat`. If this is an `Eigen::Matrix<var, R, C>` type, the output will similarly be an Eigen matrix of vars (call this a matvar for short). If the input is a `var_value<Eigen::Matrix<double, R, C>>`, the output will be a `var_value<Eigen::VectorXd>` (call this a varmat for short). Both are valid vector autodiff types in Stan.

Reply:
I think with that `singular_values` should work with varmats. Now we gotta add tests.